src/inspection/test/tests.cpp

```diff
@@ -0,0 +1,75 @@
+#include <chrono>
+#include <fstream>
+
+#include "gtest/gtest.h"
+#include "include/inspection_ros.hpp"
+#include "rclcpp/rclcpp.hpp"
+
+using testing::Eq;
+
+/**
+ * @brief test steering output
+ * 
+ */
+TEST(STEERING, steering) {
+    InspectionFunctions *new_inspection = new InspectionFunctions();
+    for (float i = 0; i < 26; i = i + 0.1) {
+        double steering = new_inspection->calculate_steering(i);
+        EXPECT_FLOAT_EQ(steering,
+    sin(i*2*M_PI/new_inspection->turning_period)*new_inspection->max_angle);
+    }
+}
+
+/**
+ * @brief test torque output. Initial speed greater than target
+ * 
+ */
+TEST(TORQUE, torque) {;
```

Comment:
Could be cool to make a test where you simulate an actual scenario, like a for with a certain number of iterations, where you send an initial speed different from the goal and in the iterations following you update the speed according to the previous output and an engineered time elapsed, to check if the controller converges

Reply:
But this one is just a mere suggestion, not too important